src/util.ts

```diff
@@ -199,21 +199,44 @@ export class ApiError extends Error {
       this.errors = errorBody.errors;
     }
 
-    const messages: string[] = [];
+    this.message = ApiError.createMultiErrorMessage(errorBody, this.errors);
+  }
+  /**
+   * Pieces together an error message by combining all unique error messages
+   * returned from a single GoogleError
+   *
+   * @private
+   *
+   * @param {GoogleErrorBody} err The original error.
+   * @param {GoogleInnerError[]} [errors] Inner errors, if any.
+   * @returns {string}
+   */
+  static createMultiErrorMessage(
+      err: GoogleErrorBody, errors?: GoogleInnerError[]): string {
+    const messages: Set<string> = new Set();
+
+    if (err.message) {
+      messages.add(err.message);
+    }
 
-    if (errorBody.message) {
-      messages.push(errorBody.message);
+    if (errors && errors.length) {
+      errors.forEach(({message}) => messages.add(message!));
+    } else if (err.response && err.response.body) {
+      messages.add(ent.decode(err.response.body.toString()));
+    } else if (!err.message) {
+      messages.add('A failure occurred during this request.');
     }
 
-    if (this.errors && this.errors.length === 1) {
-      messages.push(this.errors[0].message!);
-    } else if (this.response && this.response.body) {
-      messages.push(ent.decode(errorBody.response.body.toString()));
-    } else if (!errorBody.message) {
-      messages.push('Error during request.');
+    let messageArr: string[] = Array.from(messages);
+
+    if (messageArr.length > 1) {
+      messageArr = messageArr.map((message, i) => `    ${i + 1}. ${message}`);
```

Comment:
I think the `\t` here is the way to go.

Reply:
I'm not sure that is correct, on my local machine tabs get formatted like so

```js
const e = new Error('hi\n\thi');
console.error(e)
```

```sh
Error: hi
	hi
    at repl:1:11
    at Script.runInThisContext (vm.js:96:20)
    at REPLServer.defaultEval (repl.js:329:29)
    at bound (domain.js:396:14)
    at REPLServer.runBound [as eval] (domain.js:409:12)
    at REPLServer.onLine (repl.js:627:10)
    at REPLServer.emit (events.js:187:15)
    at REPLServer.EventEmitter.emit (domain.js:442:20)
    at REPLServer.Interface._onLine (readline.js:290:10)
    at REPLServer.Interface._line (readline.js:638:8)
```

Maybe we're better off avoiding indenting altogether

Reply:
I love it just the way it is ❤️

Reply:
Yeah, the look of the spaces is my preference as well. I thought using tabs was some type of Unix standard, but I haven't found anything to support that. So, let's just use spaces!